synapse/lib/cell.py

```diff
@@ -2584,7 +2589,10 @@ async def addHttpsPort(self, port, host='0.0.0.0', sslctx=None):
 
             sslctx = self.initSslCtx(certpath, pkeypath)
 
-        serv = self.wapp.listen(port, address=addr, ssl_options=sslctx)
+        kwargs = {
+            'xheaders': self.conf.reqConfValu('https:parse:proxy:remoteip')
```

Comment:
Does this option have any additional side effects we need to account for? If so, we could always add a `getPeerAddr()` method or something that parses the headers ourselves.

Reply:
The request protocol (which identifies if a request was made with `https` vs `http` ) is also respected if the `X-Forwarded-Proto` or `X-Scheme` headers are present.

We do not use that value in any of our handlers.